### PR TITLE
Fix compile warning: implicit type conversion

### PR DESCRIPTION
**Pull request by biojppm**
Fixes a compiler warning in Intel 2021.1 under `-Wall -Wextra -pedantic`:
```
/home/runner/work/c4core/c4core/src/c4/ext/fast_float_all.h:319:49: error: implicit conversion between floating point types of different sizes [-Werror,-Wimplicit-float-size-conversion]
constexpr static float powers_of_ten_float[] = {1e0, 1e1, 1e2, 1e3,
1e4, 1e5,
```
Also:
- Fix workflow for ubuntu 20.04
- Enable running tests under cmake cross compilation.
- This required adding the `-C <config>` argument to ctest under Windows to be able to expand the generator expression `$<TARGET_FILE:${TEST_NAME}>`.
- In turn, this required a substantial change to the VS17 workflows. This changeset grew much larger than anticipated and would have been better dealt with in a separate PR.

### CODE REVIEW

File: .github/workflows/vs17-cxx20.yml [hunk 1]
```diff
@@ -21,4 +21,4 @@ jobs:
           cd build &&
           cmake  ${{matrix.cxx}} ${{matrix.arch}}  -DCMAKE_CXX_STANDARD=20 -DFASTFLOAT_TEST=ON -DCMAKE_INSTALL_PREFIX:PATH=destination ..  &&
           cmake --build .  --verbose &&
-          ctest --output-on-failure
+          ctest --output-on-failure -C Debug
```
Comment (lemire) [inline .github/workflows/vs17-cxx20.yml:24]: This seems to effectively disable our tests. Please see CI outputs.

**Comment by lemire** [comment]
The change is uncontroversial, I think, but I don't think we want to accidentally suppress our tests under Visual Studio !

File: .github/workflows/vs17-cxx20.yml (hunk 1 above)
Comment (biojppm) [reply .github/workflows/vs17-cxx20.yml:24]: Indeed, thanks for the catch. Fix incoming.

**Comment by lemire** [comment]
It seems that the change trigger installation-test failures in CI. Hmmm...

**Comment by biojppm** [comment]
Adding the generator expression ended up requiring a substantial change to the VS17 workflows. Although initially it looked like a quick fix, this changeset grew much larger than anticipated, and would have been better dealt with in a separate PR. Sorry for the nuisance. It is working now, but if you have some concerns that this is overreaching, let me know, and I'll be happy to address those concerns, either here or in a separate PR.

**Comment by lemire** [comment]
Are you confident that our tests are still running as before? If so, let us merge.

**Comment by biojppm** [comment]
The initial error notwithstanding, the workflows that I did touch are certainly running to completion:
- [Ubuntu 20](https://github.com/fastfloat/fast_float/actions/runs/3784416378/jobs/6433709288#step:3:1)
- [VS17](https://github.com/fastfloat/fast_float/actions/runs/3784416368/jobs/6433709472#step:5:1) with [install tests as well](https://github.com/fastfloat/fast_float/actions/runs/3784416368/jobs/6433709472#step:7:1)
- [VS17-C++20](https://github.com/fastfloat/fast_float/actions/runs/3784416371/jobs/6433709728#step:5:1)
- [VS17-CLANG](https://github.com/fastfloat/fast_float/actions/runs/3784416373/jobs/6433709702#step:5:1)
- [VS17-ARM](https://github.com/fastfloat/fast_float/actions/runs/3784416380/jobs/6433709704#step:4:1) (tests cannot be run, as before)

Also, as a side effect of the changes, the VS workflows now cover both Debug and Release, which in my experience is good as it enables assertions on one end and optimizer passes on the other end, each of them bringing up specific types of issues/compiler warnings.

**Pull request merged**